### PR TITLE
Write initial draft of readme

### PR DESCRIPTION
**Pull request by Dave-Gallup**
Initial draft of README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,24 +2,47 @@
...
+## Description
 
-```
-npm i
-npm run up
-```
+An Express server/router which serves as the backend for the Databraid Slack Widget.  This is used in conjunction with the Databraid Slack SPA.  It is built to run as a container within Docker.
 
-When finished developing do:
+The Databraid Slack API handles all interaction with Slack's Event API.  It authenticates with Slack during an OAuth sign-in.  It makes requests to the Slack Event API to fetch information about a Slack team's users and channels.  It interfaces with a PostgreSQL database to store and retrieve data. It interfaces with Google's Natural Language Processor API to perform sentiment analysis on the channel's content. It is also responsible for setting up a web socket and, once the Databraid Slack SPA connects with that socket, pushing new event and sentiment information to the client.
```
Comment (joshwyatt) [inline README.md]: @Dave-Gallup I think I see a double space after one of these periods.

Comment (joshwyatt) [inline README.md]: @Dave-Gallup if there is a quick link to the Slack Event API, please provide it

Comment (joshwyatt) [inline README.md]: @Dave-Gallup if you would please also provide some link to the Google nlp.

Comment (joshwyatt) [inline README.md]: @Dave-Gallup "a web socket using Socket.IO"

File: README.md [hunk 2]
```diff
@@ -2,24 +2,47 @@
...
+
+## Usage
+
+There are a number of things that need to be in place prior to having this server function correctly.  You must:
+
+- Have a __Slack team__ to which you hold administrative privileges.
+- Set up a __Slack App__ at [api.slack.com](https://api.slack.com/) which listens for new events for that team and pushes these to the client.  You will also need to verify the server with the Slack App using the servers `/slack/events` endpoint.
+- Get a __Google Account__ and license key for Google NLP API.
```
Comment (joshwyatt) [inline README.md]: @Dave-Gallup if you can provide a link here that will be very helpful

File: README.md [hunk 3]
```diff
@@ -2,24 +2,47 @@
...
+- If this is being run locally, you will need Docker installed.
+- If this is being set up as a local development environment, you may also need to __expose your localhost__ and port using a tool such as [localtunnel](https://localtunnel.github.io/www/) or [ngrok](https://ngrok.com/download).
```
Comment (joshwyatt) [inline README.md]: "__expose your localhost__ and port over HTTPS"

File: README.md [hunk 4]
```diff
@@ -2,24 +2,47 @@
...
+
+Once these are set up, you will need to do the following to run the API:
+
+1. Create a `.env` file locally which will hold all of the variables needed to run
```
Comment (joshwyatt) [inline README.md]: "needed to run:"

File: README.md [hunk 5]
```diff
@@ -2,24 +2,47 @@
...
+    - `GOOGLE_APPLICATION_CREDENTIALS`: The filepath to the service account file containing google's API credentials.  (e.g. "./google-credentials/service-account.json")
+    - `SLACK_CLIENT_ID`: The client ID found in the _Basic Information_ section of the Slack App.
+    - `SLACK_CLIENT_SECRET`: The client secret found in the _Basic Information_ section of the Slack App.
+    - `SLACK_VERIFICATION_TOKEN`: The verification token found in the _Basic Information_ section of the Slack App.
+    - `REDIRECT_URI`: This is the URI on this API built to handle the OAuth redirect.  It will be at `<your_domain>/slack/auth/redirect`
+    - `DB_HOST`: Host for the PostgreSQL database instance being used.
```
Comment (joshwyatt) [inline README.md]: @Dave-Gallup please remove `.` from end of line for consistency.

File: README.md [hunk 6]
```diff
@@ -2,24 +2,47 @@
...
+    - `DB_PORT`: 5432
+    - `DB_USERNAME`: "databraid"
+    - `DB_PASSWORD`: your password for the database
+    - `DB_NAME`: name of the database being used.
```
Comment (joshwyatt) [inline README.md]: @Dave-Gallup please remove . from end of line for consistency.

File: README.md (hunk 1 above)
Comment (Dave-Gallup) [reply README.md]: That is common for me.  I think I've removed all of them.

Comment (Dave-Gallup) [reply README.md]: Done

Comment (Dave-Gallup) [reply README.md]: Done

Comment (Dave-Gallup) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (Dave-Gallup) [reply README.md]: I think I did this, now.

File: README.md (hunk 3 above)
Comment (Dave-Gallup) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (Dave-Gallup) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (Dave-Gallup) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (Dave-Gallup) [reply README.md]: done

**Pull request merged**